presto-hive/src/main/java/com/facebook/presto/hive/HiveS3Optimizer.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.hive.metastore.Column;
+import com.facebook.presto.hive.metastore.Partition;
+import com.facebook.presto.hive.metastore.Table;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.log.Logger;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.compress.CompressionCodec;
+import org.apache.hadoop.io.compress.GzipCodec;
+import org.apache.hadoop.mapred.InputFormat;
+import org.apache.hadoop.mapred.TextInputFormat;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+
+import static com.facebook.presto.hive.metastore.MetastoreUtil.getHiveSchema;
+import static org.apache.hadoop.hive.metastore.api.hive_metastoreConstants.META_TABLE_LOCATION;
+
+/**
+ * HiveS3Optimizer uses Amazon S3 Select to push down queries to Amazon S3. This allows presto to retrieve only a
+ * subset of data rather than retrieving the full S3 object thus improving Presto query performance.
+ */
+public class HiveS3Optimizer
+{
+    private static final Logger log = Logger.get(HiveS3Optimizer.class);
+    private static final String S3_PREFIX = "s3://";
+
+    private static Set<String> supportedSerdes = ImmutableSet.of(
+            "org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe");
+
+    private static Set<String> supportedInputFormats = ImmutableSet.of(
+            "org.apache.hadoop.mapred.TextInputFormat");
+
+    private static Set<String> supportedColumnTypes = ImmutableSet.of(
+            "boolean",
+            "int",
+            "tinyint",
+            "smallint",
+            "bigint",
+            "string",
+            "float",
+            "double",
+            "decimal",
+            "date",
+            "timestamp");
+
+    private final HiveClientConfig hiveClientConfig;
+
+    HiveS3Optimizer(HiveClientConfig hiveClientConfig)
+    {
+        this.hiveClientConfig = hiveClientConfig;
+    }
+
+    private static boolean isSerdeSupported(Properties schema)
+    {
+        String serdeName = HiveUtil.getDeserializerClassName(schema);
+
+        if (!supportedSerdes.contains(serdeName)) {
+            log.warn("Unsupported S3Select serde format found: " + serdeName);
```

Comment:
We added additional logging inside these methods because we abstracted the Hive properties (serdeName, location, etc) extraction logic in these methods. The caller doesn't extract them to keep the method working at a higher level of abstraction. 

Thus, adding logs in these helper methods provide the Presto user with insights into the root cause for failure. For instance, rather that logging "Unsupported S3Select location found" in the caller we could log "Unsupported S3Select location found: ${location}" which is helpful for debugging.

I will remove the logs from the caller. Also, will change these helper logs to debug.